testutil/node.go

```diff
@@ -0,0 +1,58 @@
+// Copyright 2019 The Swarm Authors
+// This file is part of the Swarm library.
+//
+// The Swarm library is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Lesser General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// The Swarm library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public License
+// along with the Swarm library. If not, see <http://www.gnu.org/licenses/>.
+
+package testutil
+
+import (
+	"net"
+	"testing"
+
+	"github.com/ethereum/go-ethereum/p2p/enode"
+	"github.com/ethereum/go-ethereum/p2p/enr"
+	"github.com/ethereum/go-ethereum/p2p/simulations/adapters"
+	"github.com/ethersphere/swarm/chunk"
+	"github.com/ethersphere/swarm/network"
+)
+
+// NodeConfigAtPo brute forces a node config to create a node that has an overlay address at the provided po in relation to the given baseaddr
+func NodeConfigAtPo(t *testing.T, baseaddr []byte, po int) *adapters.NodeConfig {
+	foundPo := -1
+	var conf *adapters.NodeConfig
+	for foundPo != po {
```

Comment:
This seems to be mixing enode and bzzaddresses. If you want a bzzaddress with a certain po from another address, you can also use:

```
pot.RandomAddressAt
```

Reply:
why? it creates the kademlia base address using all of this enr voodoo and checks it for the po from the supplied `baseaddr`, which is also coming out the kademlia of the reference node.i dont need just a random address, i need the whole config that is used to generate a new node so that when the node is generated, the correct address is bootstrapped into its kademlia

Reply:
The `RandomAddressAt` doesn't generate a random address, but one with a given `po` - same as what you're doing here.

ENR and Enodes are independent from BzzAddr.